pkg/synthetictests/operators.go

```diff
@@ -94,44 +94,67 @@ func testOperatorOSUpdateStaged(events monitorapi.Intervals, clientConfig *rest.
 	failThreshold := 10 * time.Minute
 
 	// Scan all OSUpdateStarted and OSUpdateStaged events, sort by node.
-	nodeOSUpdateTimes := map[string]*startedStaged{}
+	nodeNameToOSUpdateTimes := map[string]*startedStaged{}
 	for _, e := range events {
-		if strings.Contains(e.Message, "reason/OSUpdateStarted") {
-			// locator will be of the form: node/ci-op-j34hmfqt-253f3-cq852-master-1
-			_, ok := nodeOSUpdateTimes[e.Locator]
+		nodeName, _ := monitorapi.NodeFromLocator(e.Locator)
+		if len(nodeName) == 0 {
+			continue
+		}
+
+		reason := monitorapi.ReasonFrom(e.Message)
+		phase := monitorapi.PhaseFrom(e.Message)
+		switch {
+		case reason == "OSUpdateStarted":
+			_, ok := nodeNameToOSUpdateTimes[nodeName]
 			if !ok {
-				nodeOSUpdateTimes[e.Locator] = &startedStaged{}
+				nodeNameToOSUpdateTimes[nodeName] = &startedStaged{}
 			}
 			// for this type of event, the from/to time are identical as this is a point in time event.
-			ss := nodeOSUpdateTimes[e.Locator]
+			ss := nodeNameToOSUpdateTimes[nodeName]
 			ss.OSUpdateStarted = e.To
-		} else if strings.Contains(e.Message, "reason/OSUpdateStaged") {
-			// locator will be of the form: node/ci-op-j34hmfqt-253f3-cq852-master-1
-			_, ok := nodeOSUpdateTimes[e.Locator]
+
+		case reason == "OSUpdateStaged":
+			_, ok := nodeNameToOSUpdateTimes[nodeName]
 			if !ok {
-				nodeOSUpdateTimes[e.Locator] = &startedStaged{}
+				nodeNameToOSUpdateTimes[nodeName] = &startedStaged{}
 			}
 			// for this type of event, the from/to time are identical as this is a point in time event.
-			ss := nodeOSUpdateTimes[e.Locator]
+			ss := nodeNameToOSUpdateTimes[nodeName]
+			// this value takes priority over the backstop set based on the node update completion, so there's no reason
+			// to perform a check, just directly overwrite.
 			ss.OSUpdateStaged = e.To
+
+		case phase == "Update":
+			_, ok := nodeNameToOSUpdateTimes[nodeName]
+			if !ok {
+				nodeNameToOSUpdateTimes[nodeName] = &startedStaged{}
+			}
+			// This type of event indicates that an update completed. If an update completed, then we know that the latest
+			// possible time that it could have OSUpdateStaged is when the update is finished.  If we have not yet observed
+			// an OSUpdateStaged event, record this time as the final time.
+			ss := nodeNameToOSUpdateTimes[nodeName]
+			if ss.OSUpdateStaged.IsZero() {
```

Comment:
Thinking we should log a warning here clarifying what we did, and what it likely means.

Reply:
> Thinking we should log a warning here clarifying what we did, and what it likely means. 

how?

Reply:
Thought just a printf but as discussed on slack, success = no output, so that won't work.